app/controllers/profiles_controller.rb

```diff
@@ -63,11 +57,23 @@ def profile_params
   end
 
   def user_params
-    params.require(:user).permit(:display_name)
+    params.require(:user).permit(:display_name, :unique_name)
+  end
+
+  def authorize_current_user
+    redirect_to root_url unless current_user == @user
+  end
+
+  def find_user
+    @user ||=
+      if params[:unique_name]
+        User.find_by(unique_name: params[:unique_name])
+      elsif params[:user_id]
+        User.find(params[:user_id])
+      end
   end
```

Comment:
`before_action` hook for this so that we always find the user & profile before taking any controller action